tests/test_move.py

```diff
@@ -60,3 +82,91 @@ def test_move_dir(self):
         if self.preserve_time:
             dst_file2_info = dst_fs.getinfo("bar/baz.txt", namespaces)
             self.assertEqual(dst_file2_info.modified, src_file2_info.modified)
+
+
+class TestMove(unittest.TestCase):
+    def test_move_file_tempfs(self):
+        with open_fs("temp://") as a, open_fs("temp://") as b:
+            dir_a = a.makedir("dir")
+            dir_b = b.makedir("subdir")
+            dir_b.writetext("file.txt", "Content")
+            fs.move.move_file(dir_b, "file.txt", dir_a, "here.txt")
+            self.assertEqual(a.readtext("dir/here.txt"), "Content")
+            self.assertFalse(b.exists("subdir/file.txt"))
+
+    def test_move_file_fs_urls(self):
+        # create a temp dir to work on
+        with open_fs("temp://") as tmp:
+            path = tmp.getsyspath("/")
+            tmp.writetext("file.txt", "Content")
+            tmp.makedir("subdir")
+            fs.move.move_file(
+                "osfs://" + path,
+                "file.txt",
+                "osfs://" + join(path, "subdir"),
```

Comment:
Again, I think these ought to be independent TempFSes, rather than one being in a subdirectory of the other. IMHO.

Reply:
Ok, I'll look into this in the coming days, thank you for the thorough review 👍

Reply:
Ahh, true. But I think it would be better if the FSes didn't "overlap", e.g. you could do something like `"osfs://" + join(path, "subdir1")` for the first FS-URL and `"osfs://" + join(path, "subdir2")` for the second FS-URL?

Reply:
Yes sound good